### PR TITLE
HBW-38 Add substitution to static fields

### PR DESCRIPTION
**Pull request by GideonEXE**

### CODE REVIEW

File: hbw/app/models/hbw/fields/static.rb [hunk 1]
```diff
@@ -3,6 +3,20 @@ module Fields
...
+        result[:html] = prepare_string(result[:html])
+        result
+      end
+
+      def prepare_string(str)
+        str.gsub(/%{.*?}/){|s| prepare_value(s)}
+      end
+
+      def prepare_value(str)
+        task.definition['variables'].select{|v|  v['name'] == str.gsub(/[%\{\}]/, '')}.first['value']
```
Comment (sugarfree1) [inline hbw/app/models/hbw/fields/static.rb]: Two spaces:
```
|v|  v
```

Fix formatting for blocks. This is how we format them https://github.com/latera/homs/blob/master/hbw/lib/hbw/container.rb#L9

File: hbw/config/yml_api.default.yml [hunk 2]
```diff
@@ -218,6 +218,11 @@ get:
           label: Choose option
           css_class: col-xs-12
           fields:
+          - name: homsOrderDataStatic
+            type: static
+            label: TEST
+            css_class: col-xs-12 col-sm-6 col-md-4
+            html: "<h4> TEST STATIC %{homsOrderCode}</h4>"
```
Comment (sugarfree1) [inline hbw/config/yml_api.default.yml]: Format is $homsOrderCode, not %{}.

Comment (sugarfree1) [inline hbw/config/yml_api.default.yml]: Add two more test cases:
```
$homsOrderCode TEST STATIC $homsOrderCode
```
and
```
$homsOrderCode TEST STATIC $homsOrderCodeOther
```

File: hbw/app/models/hbw/fields/static.rb (hunk 1 above)
Comment (sugarfree1) [inline hbw/app/models/hbw/fields/static.rb]: BTW http://apidock.com/ruby/Enumerable/find

Comment (sugarfree1) [inline hbw/app/models/hbw/fields/static.rb]: Simplify ```prepare_value```

**Comment by GideonEXE** [comment]
UPD

File: hbw/app/models/hbw/fields/static.rb [hunk 3]
```diff
@@ -3,6 +3,26 @@ module Fields
...
       self.default_data_type = :string
 
+      def as_json
+        result = definition.symbolize_keys.dup
+        result[:html] = prepare_string(result[:html])
+        result
+      end
+
+      def prepare_string(str)
+        str.gsub(/\$\w*\b?/){ |s| prepare_value(s) }
```
Comment (sugarfree1) [inline hbw/app/models/hbw/fields/static.rb]: Nice try. Space between ```)``` and ```{``` required.

File: spec/hbw/features/static_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    expect(page).to have_content 'Orders list'
+    expect_widget_presence
+
+    click_on 'ORD-4'
+
+    expect(page).to have_content 'ORD-4'
+    expect(page).to have_content 'ORD-4 TEST STATIC ORD-4'
+    expect_widget_presence
+  end
+
```
Comment (sugarfree1) [inline spec/hbw/features/static_spec.rb]: extra white line

**Comment by GideonEXE** [comment]
UPD

**Pull request merged**